staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go

```diff
@@ -24,15 +24,17 @@ import (
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apiserver/pkg/admission"
+	serveroptions "k8s.io/apiserver/pkg/server/options"
 	"k8s.io/sample-apiserver/pkg/admission/wardleinitializer"
 	"k8s.io/sample-apiserver/pkg/apis/wardle"
 	informers "k8s.io/sample-apiserver/pkg/client/informers/internalversion"
 	listers "k8s.io/sample-apiserver/pkg/client/listers/wardle/internalversion"
 )
 
 // Register registers a plugin
-func Register(plugins *admission.Plugins) {
-	plugins.Register("BanFlunder", func(config io.Reader) (admission.Interface, error) {
+func Register(options *serveroptions.AdmissionOptions) {
+	options.RecommendedPluginOrder = append(options.RecommendedPluginOrder, "BanFlunder")
```

Comment:
Plugins appending themselves doesn't seem right… they don't know about other plugins, or what order makes sense, right?

Reply:
+1, This seems like something which should be done in k8s.io/sample-apiserver/pkg/cmd/server/start.go.